src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -181,36 +128,73 @@ public void InstallWorkloads(IEnumerable<WorkloadId> workloadIds, bool skipManif
                         Reporter.WriteLine(string.Format(LocalizableStrings.WorkloadAlreadyInstalled, string.Join(" ", previouslyInstalledWorkloads)).Yellow());
                     }
                     workloadIds = workloadIds.Concat(installedWorkloads).Distinct();
+                    workloadIds = WriteSDKInstallRecordsForVSWorkloads(workloadIds);
 
-                    var useWorkloadSets = ShouldUseWorkloadSetMode(_sdkFeatureBand, _dotnetPath);
-                    useRollback = !string.IsNullOrWhiteSpace(_fromRollbackDefinition);
-
-                    _workloadManifestUpdater.UpdateAdvertisingManifestsAsync(includePreviews, useWorkloadSets, offlineCache).Wait();
-
-                    if (useWorkloadSets)
+                    if (!_skipManifestUpdate)
                     {
-                        manifestsToUpdate = InstallWorkloadSet(context);
+                        var installStateFilePath = Path.Combine(WorkloadInstallType.GetInstallStateFolder(_sdkFeatureBand, _dotnetPath), "default.json");
+                        if (string.IsNullOrWhiteSpace(_fromRollbackDefinition) && File.Exists(installStateFilePath) && InstallStateContents.FromString(File.ReadAllText(installStateFilePath)).Manifests is not null)
+                        {
+                            //  If there is a rollback state file, then we don't want to automatically update workloads when a workload is installed
+                            //  To update to a new version, the user would need to run "dotnet workload update"
+                            //  TODO: We should also skip the update if a workload set version is pinned in install state or in global.json
+                            _skipManifestUpdate = true;
+                        }
                     }
-                    else
+
+                    RunInNewTransaction(context =>
                     {
-                        manifestsToUpdate = useRollback ? _workloadManifestUpdater.CalculateManifestRollbacks(_fromRollbackDefinition) :
-                            _workloadManifestUpdater.CalculateManifestUpdates().Select(m => m.ManifestUpdate);
-                    }
+                        if (_skipManifestUpdate)
+                        {
+                            _workloadInstaller.InstallWorkloads(workloadIds, _sdkFeatureBand, context, offlineCache);
```

Comment:
I think this raises an interesting question:
If you ask for a workload set (via global.json, say) and you also specify to skip manifest updates, what should our response be? I don't think "install the latest version" makes sense because that would violate our workload set, but we can't update other manifests, so there's no point in aligning with the workload set. Perhaps that should be an error?